Algorithms/Strings/Permutations.cs

```diff
@@ -73,19 +73,13 @@ public static bool IsAnargram(string source, string other)
                 return true;
 
             int len = source.Length;
-            // Hash set which will contains all the characters present in input source.
-            var hashSetSourceChars = new HashSet<char>();
-            var hashSetOtherChars = new HashSet<char>();
-            for (int i = 0; i < len; i++)
+            //loops and removes chars from other string so they wouldn't count the same char twice
+            for(int i = 0; i<len;i++)
             {
-                hashSetSourceChars.Add(source[i]);
-                hashSetOtherChars.Add(other[i]);
-            }
-            for (int i = 0; i < len; i++)
-            {
-                // Inputs are not Anargram if characers from *other are not present in *source.
-                if (!hashSetSourceChars.Contains(other[i])) return false;
-                if (!hashSetOtherChars.Contains(source[i])) return false;
+                int index = other.IndexOf(source[i]);
+                if(index==-1)
+                    return false;
+                other.Remove(index);
```

Comment:
Absolutly critical code!

1. Remove(index) - remove From to End. You are drop all symbols from right.
2. String immunable. "other" object it not changed.
Example: `other = other.Remove(index, 1);`
But this bad too - immunable string operation = bad performance, memory and etc. problem.

StringPermutationTests discredited. This test case must will be fail.
`
	var one = "dbcdefg";
	var two = "dabcgfe";
	Assert.True(Permutations.IsAnargram(one, two) == true);
`